src/tracer.c

```diff
@@ -227,8 +227,18 @@ handle_syscall(PROCESS_INFO *pi, const struct ptrace_syscall_info *entry,
 	    // int flags);
 	    record_process_start(pi->pid, pi->outname);
 	    break;
-	default:
-	    return;
+	case SYS_fork:
+	    // pid_t fork(void);
+	    record_process_create(pi->outname, find(exit->exit.rval)->outname);
```

Comment:
How do you expect to `find()` the new process? Where was it entered in `pinfo` ?

Reply:
@zvr The order of stops upon one of those syscalls is:

first stop: `PTRACE_SYSCALL_ENTRY`
second stop: stop from setting `PTRACE_O_TRACEFORK`, `PTRACE_O_TRACEVFORK`, `PTRACE_O_TRACECLONE`
last stop: `PTRACE_SYSCALL_EXIT`


Is that documented anywhere? Perhaps it is.
`ptrace(2)`'s manual states:

**Stop the tracee at the next fork(2) and automatically
                     start tracing the newly forked process, which will start
                     with a SIGSTOP, or PTRACE_EVENT_STOP if PTRACE_SEIZE was
                     used.  A waitpid(2) by the tracer will return a status
                     value such that

                       status>>8 == (SIGTRAP | (PTRACE_EVENT_FORK<<8))

                     The PID of the new process can be retrieved with
                     PTRACE_GETEVENTMSG.**
                
Now you might argue that this doesn't assume that the stop is caused before `fork(2)` exits. But i am pretty sure the existence of this alternative option does:

**PTRACE_O_TRACEVFORKDONE (since Linux 2.5.60)
                     Stop the tracee at the completion of the next vfork(2).
                     A waitpid(2) by the tracer will return a status value
                     such that

                       status>>8 == (SIGTRAP | (PTRACE_EVENT_VFORK_DONE<<8))

                     The PID of the new process can (since Linux 2.6.18) be
                     retrieved with PTRACE_GETEVENTMSG.**
                     
And thus i made the assumption that the new process will not only exist, but it will also be at the top of the array(the last element). Althought taking into considerations that there might be some weird scenarios where this isn't the case, i just used `find(3)`.

Reply:
@zvr `tracer_main(3)` at the `SIGSTOP` switch case.

Reply:
And oh well why did i add those three syscall handlers instead of handling it at `SIGSTOP` you might ask? Well it's because we can't really know the one who created the new process at the `SIGSTOP` switch case.

What we could do tho is use the fact that the one who created the new process is also stopped with a `SIGTRAP` as stated in the manual and add a handler in `tracer_main` that does just that. Would we prefer that?